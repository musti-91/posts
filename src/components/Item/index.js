// @flow
import React from "react";
import styled from "styled-components";

import pen from "../../styles/edit.svg";
import deleteIcon from "../../styles/delete.svg";
import Icon from "../Icon";

type Props = {
  title: string,
  id: string,
  body: string,
  checked?: boolean,
  onClick: (id: number | string) => void,
  onDelete: (id: number | string) => void,
  onChecked: (id: number | string) => void
};

const styles = {
  color: "#447",
  marginBottom: ".5em",
  backgroundColor: "grey",
  padding: ".4em",
  fontSize: "1em",
  borderRadius: ".3em",
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
  span: {
    width: "60%"
  }
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

const Item = ({ title, id, body, onClick, onChecked, onDelete }: Props) => {
  return (
    <div id={`item-${id}`} style={styles}>
      <span style={styles.span} onClick={() => onClick(id)}>
        {title}
        {body && <p>{body}</p>}
      </span>
      <Wrapper>
        <Icon
          src={deleteIcon}
          description={`delete-${title}`}
          onClick={() => onDelete(id)}
        />
        <Icon
          src={pen}
          description={"alt-" + title}
          onClick={() => onChecked(id)}
        />
      </Wrapper>
    </div>
  );
};

export default Item;
