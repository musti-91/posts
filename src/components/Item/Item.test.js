/**
 * @jest-environment jsdom
 */
/* eslint-disable react/display-name */
import React from "react";
import { shallow } from "enzyme";
/**
 *import  Component
 */
import Item from "./index";
/**
 * props
 */
const props = {};
/**
 * Tests: Item Component
 */
describe("Item Component", () => {
  /**
   * Snapshot:[default]
   */
  test("should match default snapshot", () => {
    expect.assertions(1);

    const wrapper = shallow(<Item {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
