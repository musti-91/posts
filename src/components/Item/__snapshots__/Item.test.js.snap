// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component should match default snapshot 1`] = `
<div
  id="item-undefined"
  style={
    Object {
      "backgroundColor": "grey",
      "borderRadius": ".3em",
      "color": "#447",
      "display": "flex",
      "flexDirection": "row",
      "fontSize": "1em",
      "justifyContent": "space-between",
      "marginBottom": ".5em",
      "padding": ".4em",
      "span": Object {
        "width": "60%",
      },
    }
  }
>
  <span
    onClick={[Function]}
    style={
      Object {
        "width": "60%",
      }
    }
  />
  <styled.div>
    <Icon
      description="delete-undefined"
      onClick={[Function]}
      src="delete.svg"
    />
    <Icon
      description="alt-undefined"
      onClick={[Function]}
      src="edit.svg"
    />
  </styled.div>
</div>
`;
